### PR TITLE
Introduce MultiCommand class for cmds operating on multiple todos.

### LINKED ISSUE #14: Code refactoring for commands operating on multiple todo IDs
As it was said under #8:
> Overall, the function has become a bit long and deeply nested (...)

I thought that it would be nice to have one common class for all commands dealing with multiple todo IDs (`do`, `del`, `pri`, `postpone` at the moment). I've implemented it here: https://github.com/mruwek/topydo/commits/multi-todos-cmds-refactoring (sorry about long branch name :P).

At this moment `MultiCommand` class has only two methods, but I believe in future we can put there even more reusable code (like some magic function for selecting all todos with given project or context). I didn't open new pull request yet, because I'm not sure if code is mature enough (naming convention etc.) and if You really perceive it as a step in the right direction. I'm open to any critique,

In near future `edit` and `depri` could also benefit from this,

### PR DESCRIPTION
**Pull request by mruwek**
This fixes #14

@bram85 I've implemented your suggestions and added new `execute_multi_specific()` method to reduce indentation and overall code duplication in `execute()`. Tests are still valid :)

### CODE REVIEW

File: topydo/lib/MultiCommand.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        self.invalid_numbers = []
+
+    def get_todos(self, p_numbers):
+        for number in p_numbers:
+            try:
+                self.todos.append(self.todolist.todo(number))
+            except InvalidTodoException:
+                self.invalid_numbers.append(number)
+
+    def catch_todo_errors(self):
```
Comment (bram85) [inline topydo/lib/MultiCommand.py]: Function should be private (start with `_`).

File: topydo/lib/MultiCommand.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                 p_out=lambda a: None,
+                 p_err=lambda a: None,
+                 p_prompt=lambda a: None):
+        super(MultiCommand, self).__init__(
+            p_args, p_todolist, p_out, p_err, p_prompt)
+
+        self.todos = []
+        self.invalid_numbers = []
+
+    def get_todos(self, p_numbers):
```
Comment (bram85) [inline topydo/lib/MultiCommand.py]: A small docstring that describes what this function does would be helpful.

File: topydo/lib/MultiCommand.py (hunk 1 above)
Comment (bram85) [inline topydo/lib/MultiCommand.py]: Please also add a docstring here.

**Comment by bram85** [comment]
Looks good overall, will merge it in if you deal with the minor remarks (doesn't affect any functionality, just style things).

**Comment by mruwek** [comment]
Updated and rebased. Changelog:
1. Add docstrings for `MultiCommand.catch_errors()` and `MultiCommand.get_todos()`
2. Turn `MultiCommand.catch_errors()` into private method --> `MultiCommand._catch_errors()`
3. Define `MultiCommand.execute_multi_specific()` in `MultiCommand.MultiCommand` itself and add docstring there.

**Comment by bram85** [comment]
Thanks for doing this.

**Pull request merged**